### PR TITLE
Docs refactor and updates

### PR DESCRIPTION
**Pull request by drdavella**
This PR refactors and updates our docs to put a greater emphasis on triage and tool remediation. It is intended as a starting point for additional docs update. The docs are intended to answer the following questions of a potential customer:

* What does Pixeebot do?
* What tools/languages/rules does Pixeebot support?
* How do I get started?

Probably some additional cross-links between sections would be useful.

I did not attempt to improve the styling at all; this might be worth revisiting at a later date.

### CODE REVIEW

**Review by terranosaurus — COMMENTED** [review]
- Landing page makes sense
    - Get started tile: Should we call out GitHub app so explicitly? On the linked Installation page do we need to mention the GH action or how to rig up for other SCMs like Bitbucket?
- This has bugged me for awhile but I never got back to it: the Pixee logo links to our marketing website but in the user platform it’s how a user returns to the landing page (dashboard). Here in Docs, there’s no way to get back to the landing page other than via the back button because of this. Many other well-known technical docs sites don’t link to their respective marketing sites in this way - see Stripe, GitHub, GitLab, Docker, etc. That being said, they usually have a DOCS label with the logo, which I can easily create.  Personally I feel our Pixee logo should link back to the docs landing page.
- Open Pixee page - agree with Arshan’s notes re: GH-centric and overall story right now, but otherwise was going to suggest we at least rename it - Open Pixee was confusing to me at first. I mean it doesn’t **not** make sense, but feels awkward somehow IMO. Could just say simply “Open source”
- User guide - I feel like this page should mirror and have basic coverage of everything we call out initially in the Introduction, so should have headers and blurbs for Continuous improvement, PR improvement, and Triage. Can follow up with more ideas for that separately.
- I think the order of side nav should be updated to:
    - Intro
    - Install
    - User guide (could also see the case for Preferences first)
    - Preferences
    - Tool fixes
    - Core codemods
    - Open source
    - FAQs
    - Release notes
- Sentence case updates -  I’ll do that separately after this gets merged

**Review by nahsra — APPROVED** [review]

File: docs/code-scanning-tools/overview.md
Comment (sip49) [inline docs/code-scanning-tools/overview.md:1]: Shouldn't we also call out Node?

File: docs/code-scanning-tools/sonar.md
Comment (sip49) [inline docs/code-scanning-tools/sonar.md:1]: I generally agree with the "Pixee" vs "pixeebot" usage. However in this file, we are specifically referring to the Github app in a few places, so I assume those should be "Pixeebot" not Pixee, right?

File: docs/code-scanning-tools/sonarqube.md
Comment (sip49) [inline docs/code-scanning-tools/sonarqube.md:1]: same comments as sonarcloud file

**Review by sip49 — COMMENTED** [review]
Left a few comments throughout and also made some commits inline.

File: docs/code-scanning-tools/overview.md
Comment (sip49) [reply docs/code-scanning-tools/overview.md:1]: added

**Pull request merged**